src/hooks/replace-location.ts

```diff
@@ -16,15 +22,20 @@ export const replaceLocation = (): void => {
 	jest.spyOn(locationMock, "replace").mockName("window.location.replace");
 
 	// Add the property to the Window
-	// - Only some JSDOM versions support `delete` and `set`, so we use `Object.defineProperty`
+	// - Only some JSDOM versions support `delete` and `set`, and the browser allows for a setter to update the location
+	//   so we use `Object.defineProperty`
 	Object.defineProperty(
 		window,
 		"location",
 		{
-			value: locationMock,
+			get () {
+				return locationMock;
+			},
+			set (value: string) {
+				locationMock.href = value;
+			},
```

Comment:
I didn't implement the setter on the first go of this library. Mostly because I don't use it instead of the many other options like `window.location.href =` and forgot! TypeScript's typings are also borked here, so maybe I am not alone lol.

This is technically a breaking change, so I am glad it can be lumped together with `window.history` support